virttest/libvirtd_decorator.py

```diff
@@ -0,0 +1,126 @@
+"""
+A decorator utility functions to apply libvirtd functions.
+
+Copyright: Red Hat Inc. 2020
+"""
+import logging
+import os
+import re
+from virttest import data_dir
+
+from avocado.utils import process
+from avocado.utils import path
+from avocado.utils import astring
+
+
+try:
+    path.find_command("libvirtd")
+    LIBVIRTD = "libvirtd"
+except path.CmdNotFoundError:
+    LIBVIRTD = "virtqemud"
+
+
+def get_libvirtd_split_enable_bit():
+    base_cfg_path = os.path.join(data_dir.get_shared_dir(), 'cfg', 'base.cfg')
+    if os.path.isfile(base_cfg_path):
+        with open(base_cfg_path, 'r') as base_file:
+            for line in base_file:
+                if 'enable_split_libvirtd_feature' in line and 'yes' in line:
+                    return True
+    else:
+        logging.info("CAN NOT find base.cfg file")
+    return False
+
+
+def get_libvirt_version_compare(major, minor, update, session=None):
+    """
+    Determine/use the current libvirt library version on the system
+    and compare input major, minor, and update values against it.
+    If the running version is greater than or equal to the input
+    params version, then return True; otherwise, return False.
+
+    This is designed to handle upstream version comparisons for
+    test adjustments and/or comparisons as a result of upstream
+    fixes or changes that could impact test results.
+
+    :param major: Major version to compare against
+    :param minor: Minor version to compare against
+    :param update: Update value to compare against
+    :param session: Shell session on remote host
+    :return: True if running version is greater than or
+                  equal to the input libvirt version
+    """
+    LIBVIRT_LIB_VERSION = 0
+
+    func = process.system_output
+    if session:
+        func = session.cmd_output
+
+    libvirt_ver_cmd = "%s -V" % LIBVIRTD
+    logging.warn(libvirt_ver_cmd)
+    try:
+        regex = r'%s\s*.*[Ll]ibvirt.*\s*' % LIBVIRTD
+        regex += r'(\d+)\.(\d+)\.(\d+)'
+        lines = astring.to_text(func(libvirt_ver_cmd)).splitlines()
```

Comment:
When the LIBVIRTD cmd does not exists this raises the `FileNotFoundError` exception. I can see 2 options, one is to add this extra exception to the try/except block or to detect not just libvirtd but also virtqemud and in case it's missing set it to None and treat all places where it's used to return early when LIBVIRTD is not available.

Reply:
code updated to return early if LIBVIRTD or virtqemud both don't exists